code/datums/components/usb_port.dm

```diff
@@ -65,6 +73,13 @@
 	for(var/obj/item/circuit_component/component as anything in circuit_components)
 		component.register_usb_parent(parent)
 
+	if(extra_registration_callback)
```

Comment:
What's the point of this? Why can't you call this at the same time you're adding the USB? This feels like bad separation of concerns

Reply:
Because loading a circuit from json can cause the usb port component to unregister and then reregister itself.